.github/workflows/nix.yml

```
@@ -0,0 +1,20 @@
+name: Nix derivation checks
+
+on:
+  push:
+    paths: ["mix.lock"]
+
+jobs:
+  auto-update-nix-hash:
+    runs-on: ubuntu-latest
+    name: Auto update Nix hash
+    steps:
+    - uses: actions/checkout@v4
+    - uses: DeterminateSystems/nix-installer-action@main
+    - uses: DeterminateSystems/magic-nix-cache-action@main
+    - run: |
+        nix build --no-link .#__fodHashGen 2>&1 | awk '/gen:/ { print $2 }' > nix/hash
+        git config user.email ""
+        git config user.name "GitHub Action Bot"
+        git commit -m 'chore: update Nix FOD hash' nix/hash
```

Because we don't enforce conventional commits, I think I'd prefer this without the prefix.

```suggestion
        git commit -m 'Update Nix FOD hash' nix/hash
```